### PR TITLE
[issue-67] unit tests for FlinkPravegaWriter

### LINKED ISSUE #67: Develop a unit test for FlinkPravegaWriter
**Problem description**
The `FlinkPravegaWriter` has a pair of existing tests, `FlinkPravegaWriterTest` and `FlinkExactlyOncePravegaWriterTest`, that are best understood as integration tests.   Overall coverage is low.  

**Suggestions for an improvement**
- Develop a unit test that uses a mock Pravega event writer to fully exercise the `FlinkPravegaWriter`.
- Consolidate the existing tests into a test called `FlinkPravegaWriterITCase`.

### PR DESCRIPTION
**Pull request by EronWright**
Closes #67 
**Change log description**
- consolidate existing writer tests into FlinkPravegaWriterITCase
- develop a test harness for sink functions
- develop new unit tests for FlinkPravegaWriter
- adjust checkstyle to allow the use of mockito
- make internal fields and classes of FlinkPravegaWriter visible for testing
- fix style check failures

**Purpose of the change**
- improve test coverage of FlinkPravegaWriter

**How to verify it**
N/A

### CODE REVIEW

**Comment by EronWright** [comment]
```

File: src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterITCase.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                30 * 1000);
+        pravegaWriter.setPravegaWriterMode(PravegaWriterMode.EXACTLY_ONCE);
+
+        env
+                .addSource(new ThrottledIntegerGeneratingSource(numElements))
+                .map(new FailingMapper<>(numElements / 2))
+                .addSink(pravegaWriter).setParallelism(2);
+
+        final long executeStart = System.nanoTime();
+        System.out.println(env.getExecutionPlan());
```
Comment (fpj) [inline src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterITCase.java]: Should we use slf4j?

File: src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterITCase.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Integration tests for {@link FlinkPravegaWriter}.
+ */
+@Slf4j
+public class FlinkPravegaWriterITCase {
```
[the remark this answers is not in the document]
Comment (EronWright) [reply src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterITCase.java]: Sorry, 'ITCase' is a convention in Flink to designate integration tests.

File: src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterTest.java [hunk 3]
```diff
@@ -9,193 +9,598 @@
...
+        FlinkPravegaWriter.AbstractInternalWriter internalWriter = mock(FlinkPravegaWriter.AbstractInternalWriter.class);
+        Mockito.doReturn(internalWriter).when(sinkFunction).createInternalWriter();
+
+        try {
+            try (StreamSinkOperatorTestHarness<Integer> testHarness = createTestHarness(sinkFunction)) {
+                testHarness.open();
+                verify(internalWriter).open();
+
+                // verify that exceptions don't interfere with close
+                Mockito.doThrow(new IntentionalRuntimeException()).when(internalWriter).close();
```
[the remark this answers is not in the document]
Comment (EronWright) [reply src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterTest.java]: The intention here is to exercise the try..catch blocks within `FlinkPravegaWriter::close()` that ensure that everything is definitely closed (by suppressing the exceptions).   The mock statement causes the internal writer to throw upon `close`.

**Pull request merged**